### DISCUSSION TITLE
A trigger can read palette RGBA value?

### DISCUSSION THREAD

**Discussion by beterhans** [ikemen-engine/Ikemen-GO#2205] [Feature Request]
Im studying Eiton's Akiha

Found he hard coded the projectile effect in Akiha-sys.zss like following
```
else if palNo = 8 {	
		#d0
		MapSet{map:"eff_0_r";value:255}
		MapSet{map:"eff_0_g";value:114}
		MapSet{map:"eff_0_b";value:170}
		
		#f1
		MapSet{map:"eff_1_r";value:256}
		MapSet{map:"eff_1_g";value:0}
		MapSet{map:"eff_1_b";value:200}
		
		#f4
		MapSet{map:"eff_2_r";value:256}
		MapSet{map:"eff_2_g";value:0}
		MapSet{map:"eff_2_b";value:256}
		
		#f2
		MapSet{map:"eff_3_r";value:256}
		MapSet{map:"eff_3_g";value:128}
		MapSet{map:"eff_3_b";value:256}
		
		#f0
		MapSet{map:"eff_4_r";value:256}
		MapSet{map:"eff_4_g";value:256}
		MapSet{map:"eff_4_b";value:256}
		
		#f8
		MapSet{map:"eff_5_r";value:256}
		MapSet{map:"eff_5_g";value:0}
		MapSet{map:"eff_5_b";value:256}
```

initally I though it was based on the Palette I selected, since in official palette, there is color in the botton to decide what color should be the effect to use.

I wonder if there could be a trigger to return the RGBA value of a position of the palette will be great.

### Describe the solution you'd like

Like PalNo rutrn the number of current Pal

PalColor(-1,-1,255,r) will return current player's current palette's 255 position's red value
PalColor(1,12,36,a) will return Pal 1,12 's position 36's alpha value? which doesn't care current pal.

### Search terms

palette

**Comment by potsmugen** [comment]
Such a trigger is probably doable but it may be a little weird on its own. What I mean is it sounds like something that ought to be part of a larger feature like being able to change specific colors (imagine Demitri's aura), applying PalFX only to some color indexes, etc.
